src/renderer/components/chat-window.tsx

```diff
@@ -106,10 +115,16 @@ export function ChatWindow({ id }: { id?: string }) {
 
       if (file.name.endsWith('.pdf')) {
         setRunningEmbeddings(true)
+
         await transformersManager.embedDocument(file.path)
         await filesManager.loadFiles()
         historyManager.changeThreadFilePath(id, file.path)
+
         setRunningEmbeddings(false)
+        toast.success(`Embeddings generated for ${file.name}`)
+
+        // Rely on useState to refocus the textarea
+        setShouldRefocusTextarea(true)
```

Comment:
Does doing the check + focus in here (what you are doing in the useEffect) work?
```
if (shouldRefocusTextarea && textAreaInputRef.current) textAreaInputRef.current.focus()
```
To avoid the extra effect?

Reply:
Yes, that'd work!